feat: enhance global chat interface with modern UI/UX improvements

Ryuga-17 opened the pull request:
Features:
- Modern responsive design with mobile-first approach
- Enhanced dark/light mode support with better contrast
- Consistent border radius throughout the interface
- Improved message bubbles with gradient backgrounds
- Emoji reactions and quick reaction buttons
- Typing indicators with animated dots
- Scroll-to-bottom button with smooth animations
- Enhanced empty state with welcome messaging
- Better timestamp display (relative time)
- Message status indicators and character counter

UI/UX Improvements:
- Rounded corners and modern shadow system
- Hover effects and scale animations
- Better color contrast for accessibility
- Responsive layout for all screen sizes
- Enhanced input area with emoji picker
- Improved header with online users counter
- Better message styling for own vs others' messages

Responsive Design:
- Mobile-optimized touch targets
- Adaptive text sizes and spacing
- Flexible layouts for tablet and desktop
- Touch-friendly interactions

Technical Improvements:
- Smooth scroll behavior
- Auto-scroll to bottom functionality
- Enhanced error handling
- Better loading states
- Improved accessibility

Fixes: Chat interface width and responsiveness issues
Resolves: Modern UI/UX enhancements for better user experience

Ryuga-17 committed:
feat: enhance global chat interface with modern UI/UX improvements

✨ Features:
- Modern responsive design with mobile-first approach
- Enhanced dark/light mode support with better contrast
- Consistent border radius throughout the interface
- Improved message bubbles with gradient backgrounds
- Emoji reactions and quick reaction buttons
- Typing indicators with animated dots
- Scroll-to-bottom button with smooth animations
- Enhanced empty state with welcome messaging
- Better timestamp display (relative time)
- Message status indicators and character counter

🎨 UI/UX Improvements:
- Rounded corners and modern shadow system
- Hover effects and scale animations
- Better color contrast for accessibility
- Responsive layout for all screen sizes
- Enhanced input area with emoji picker
- Improved header with online users counter
- Better message styling for own vs others' messages

📱 Responsive Design:
- Mobile-optimized touch targets
- Adaptive text sizes and spacing
- Flexible layouts for tablet and desktop
- Touch-friendly interactions

🔧 Technical Improvements:
- Smooth scroll behavior
- Auto-scroll to bottom functionality
- Enhanced error handling
- Better loading states
- Improved accessibility

Fixes: Chat interface width and responsiveness issues
Resolves: Modern UI/UX enhancements for better user experience

Ryuga-17 commented:
@PriyanshuShekhar10 can you review the changes.

PriyanshuShekhar10 commented:
```build issue : 
18:01:17.586 
18:01:17.587 
Failed to compile.
18:01:17.587 
18:01:17.587 
./app/(product)/components/globalchat.tsx
18:01:17.587 
28:23  Error: 'setOnlineUsers' is assigned a value but never used.  @typescript-eslint/no-unused-vars
18:01:17.587 
227:9  Error: 'getReactionCount' is assigned a value but never used.  @typescript-eslint/no-unused-vars
18:01:17.587 
18:01:17.587 
info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
18:01:17.624 
Error: Command "npm run build" exited with 1
```

please run `npm run build` and fix the issues that are occuring

Ryuga-17 committed:
fix: remove unused variables causing linting errors

- Remove unused setOnlineUsers variable
- Remove unused getReactionCount function
- Fix TypeScript/ESLint compilation errors

Ryuga-17 commented:
@PriyanshuShekhar10 done.
